test/test_sbom.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (C) 2021 Anthony Harrison
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import os
+import re
+
+import pytest
+
+from cve_bin_tool.sbom_manager import Remarks, SBOMManager
+from cve_bin_tool.util import ProductInfo
+
+
+class TestSBOM:
+    SBOM_PATH = os.path.join(os.path.abspath(os.path.dirname(__file__)), "sbom")
+    PARSED_SBOM_DATA = {
+        ProductInfo(vendor="gnu", product="glibc", version="2.11.1"): {
+            "default": {"remarks": Remarks.NewFound, "comments": "", "severity": ""},
+            "paths": {""},
+        }
+    }
+
+    @pytest.mark.parametrize(
+        "filepath",
+        (os.path.join(SBOM_PATH, "nonexistent.spdx.json"),),
+    )
+    def test_nonexistent_file(self, filepath):
+        sbom_engine = SBOMManager(filepath)
+        assert sbom_engine.scan_file() == {}
+
+    @pytest.mark.parametrize(
+        "filename",
+        ((os.path.join(SBOM_PATH, "bad.csv")),),
+    )
+    def test_invalid_file(self, filename):
+        sbom_engine = SBOMManager(filename)
+        assert sbom_engine.scan_file() == {}
+
+    @pytest.mark.parametrize(
+        "filename, spdx_parsed_data",
+        (
+            (os.path.join(SBOM_PATH, "spdx_test.spdx"), PARSED_SBOM_DATA),
+            (os.path.join(SBOM_PATH, "spdx_test.spdx.rdf"), PARSED_SBOM_DATA),
+            (os.path.join(SBOM_PATH, "spdx_test.spdx.json"), PARSED_SBOM_DATA),
+            (os.path.join(SBOM_PATH, "spdx_test.spdx.xml"), PARSED_SBOM_DATA),
+            (os.path.join(SBOM_PATH, "spdx_test.spdx.yml"), PARSED_SBOM_DATA),
+            (os.path.join(SBOM_PATH, "spdx_test.spdx.yaml"), PARSED_SBOM_DATA),
+        ),
+    )
+    def test_valid_spdx_file(self, filename, spdx_parsed_data):
```

Comment:
```suggestion
    def test_valid_spdx_file(
        self, filename: str, spdx_parsed_data: DefaultDict[ProductInfo, TriageData]
    ):
```
and similar in other tests